crates/oxc_macros/src/ast_node.rs

```diff
@@ -56,10 +56,11 @@ fn modify_enum(item: &mut ItemEnum) -> NodeData {
     // add the dummy variant
     item.variants.insert(0, parse_quote!(Dummy));
     // add explicit discriminants to all variants
-    item.variants
-        .iter_mut()
-        .enumerate()
-        .for_each(|(i, var)| var.discriminant = Some((parse_quote!(=), parse_quote!(#i as u8))));
+    item.variants.iter_mut().enumerate().for_each(|(i, var)| {
+        #[allow(clippy::cast_possible_truncation)]
+        let i = i as u8;
+        var.discriminant = Some((parse_quote!(=), parse_quote!(#i)));
```

Comment:
When writing this, I checked this line, and the compiler optimizes the cast away after macro expansion; It expands to `Variant = 1 as u8` and the compiler does the cast in comptime. I've kept it that way for better readability.

Was there any issues with it? I'm just curious.